matomo_5: delete old unused files

laalsaas opened the pull request:
PL-132236

@flyingcircusio/release-managers

## Release process

Impact:

Changelog: (not yet ready to use, we are waiting for upstream commits to update the matomo package in our nixpkgs fork)

### PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [x] ticket state set to *Pull request ready*
- [x] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [x] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - pull in latest matomo version
- [x] Security requirements tested? (EVIDENCE)
  - tested upgrade from matomo 4 to matomo 5 on pntest00
  - tested setting up new matomo 5

Commit:
matomo5: fix upgrade issues

dpausp commented on nixos/services/matomo.nix:
```diff
@@ -334,6 +556,19 @@ in {
 
+              ${lib.optionalString (lib.versionAtLeast cfg.package.version "5.0.0") ''
+              if [[ -f ${webrootDir}/plugins/CoreConsole/Commands/GenerateAngularDirective.php ]]; then
+                echo "Deleting leftover core plugin file from matomo 4..."
```
```suggestion
                echo "Deleting leftover core plugin files from matomo 4..."
```

dpausp commented on nixos/services/matomo.nix:
```diff
@@ -334,6 +556,19 @@ in {
+              ''}
+
+              ${lib.optionalString (lib.versionAtLeast cfg.package.version "5.0") ''
+                echo "Deleting unnecessary files"
```
Can be removed, we already said above that we are deleting old stuff

dpausp commented on nixos/services/matomo.nix:
```diff
@@ -98,6 +98,228 @@ let
     '';
   };
 
+  # This script was generated by copying recommended commands from the Matomo web ui
```
I'd like to see this script as a separate file as it is quite long and hard to read inline.

Commit:
matomo: remove package from path of setup unit

The only binary provided by the matomo package is matomo-console. This
will not work, our matomo directory is /var/lib/matomo, whereas the
matomo-console from the package will try to write in the nix store. We
have our own matomoConsole package, which provides a modified version of
matomoConsole that will use /var/lib/matomo.
